root/entity/Tags.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import chooseLayoutComponent from '../utility/chooseLayoutComponent';
+import {l} from '../static/scripts/common/i18n';
+import {withCatalystContext} from '../context';
+import {MainTagEditor} from '../static/scripts/common/components/TagEditor';
+
+
+type Props = {|
+  +$c: CatalystContextT,
+  +allTags: $ReadOnlyArray<AggregatedTagT>,
+  +entity: CoreEntityT,
+  +lastUpdated: string,
+  +moreTags: boolean,
+  +userTags: $ReadOnlyArray<UserTagT>,
+|};
+
+const Tags = ({
+  $c,
+  allTags,
+  displayGenres,
+  displayTags,
+  entity,
+  lastUpdated,
+  moreTags,
+  userTags,
+}: Props) => {
+  const entityType = entity.entityType;
+  const LayoutComponent = chooseLayoutComponent(entityType);
+
+  return (
+    <LayoutComponent page="tags" releaseGroup={entity} title={l('Tags')}>
```

Comment:
`releaseGroup={entity}` will be incorrect for other entity types, though. I suggest that we just modify all the entity-specific layout components to accept an `entity` prop. You can still alias it as `releaseGroup` etc. in the props destructuring:

```JavaScript
const ReleaseGroupLayout = ({
  entity: releaseGroup, // prop "entity" is assigned to "releaseGroup" in the function body
  ...otherProps
}) => { /* ... */ };
```

Reply:
(Also, I was wondering why Flow didn't complain about that and realized `chooseLayoutComponent` can literally only return `ReleaseGroupLayout` at present, so it out-smarted me.)